[ci][cutlass backend] Add ci for cutlass backend tests

henrylhtsang opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #156626

redo of https://github.com/pytorch/pytorch/pull/156136

Differential Revision: [D77327309](https://our.internmc.facebook.com/intern/diff/D77327309)

I want to try land the full version first. If the ci is taking too long, we can revert back to only testing for a few names.
```
 -k 'test_max_autotune_cutlass_backend_regular_mm and not test_max_autotune_cutlass_backend_regular_mm_streamk' 
```

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @chenyang78 @kadeng @muchulee8 @amjames @chauhang @aakhundov @coconutruben

henrylhtsang committed:
[ci][cutlass backend] Add ci for cutlass backend tests

[ghstack-poisoned]

henrylhtsang committed:
Update on "[ci][cutlass backend] Add ci for cutlass backend tests"


redo of https://github.com/pytorch/pytorch/pull/156136

[ghstack-poisoned]

henrylhtsang committed:
Update on "[ci][cutlass backend] Add ci for cutlass backend tests"


redo of https://github.com/pytorch/pytorch/pull/156136

[ghstack-poisoned]

henrylhtsang committed:
Update on "[ci][cutlass backend] Add ci for cutlass backend tests"


redo of https://github.com/pytorch/pytorch/pull/156136

[ghstack-poisoned]

henrylhtsang committed:
Update on "[ci][cutlass backend] Add ci for cutlass backend tests"


redo of https://github.com/pytorch/pytorch/pull/156136

[ghstack-poisoned]

henrylhtsang committed:
Update on "[ci][cutlass backend] Add ci for cutlass backend tests"


redo of https://github.com/pytorch/pytorch/pull/156136

[ghstack-poisoned]

henrylhtsang committed:
Update on "[ci][cutlass backend] Add ci for cutlass backend tests"


redo of https://github.com/pytorch/pytorch/pull/156136

Differential Revision: [D77327309](https://our.internmc.facebook.com/intern/diff/D77327309)

[ghstack-poisoned]

clee2000 commented on .ci/pytorch/test.sh:
```diff
@@ -333,6 +333,14 @@ test_h100_distributed() {
+  folder_name="third_party/cutlass/python"
+  start_dir="."
+  folder_path=$(find "$start_dir" -type d -path "$folder_name" -print -quit)
+  TORCHINDUCTOR_CUTLASS_DIR=$relative_path python test/run_test.py --include inductor/test_cutlass_backend -k 'test_max_autotune_cutlass_backend_regular_mm and not test_max_autotune_cutlass_backend_regular_mm_streamk' $PYTHON_TEST_EXTRA_OPTION --upload-artifacts-while-running
```
I don't see where `relative_path` is defined?

henrylhtsang replied:
> I don't see where `relative_path` is defined?

sure, but if you look at the logs:
```
2025-06-25T20:57:47.9637124Z + folder_name=third_party/cutlass/python
2025-06-25T20:57:47.9637360Z + start_dir=.
2025-06-25T20:57:47.9637622Z ++ find . -type d -path third_party/cutlass/python -print -quit
2025-06-25T20:57:48.3304017Z + folder_path=
2025-06-25T20:57:48.3304497Z + TORCHINDUCTOR_CUTLASS_DIR=
2025-06-25T20:57:48.3305760Z + python test/run_test.py --include inductor/test_cutlass_backend -k 'test_max_autotune_cutlass_backend_regular_mm and not test_max_autotune_cutlass_backend_regular_mm_streamk' --upload-artifacts-while-running
```
folder_path is still empty?

clee2000 replied:
`find . -type d -path third_party/cutlass/python -print -quit` doesn't print anything for me when I run locally, you probably want `./third_party/cutlass/python` or some wildcards

I've also historically had some issues with find where I think it was buffering output or something so be careful

henrylhtsang replied:
@clee2000  in the end looks like I need absolute path LOL

the path that works `TORCHINDUCTOR_CUTLASS_DIR=/var/lib/jenkins/workspace/third_party/cutlass` saw from raw logs

huydhn replied:
I was about to say you could use $GITHUB_WORKSPACE, but it seems to hard code to `/var/lib/jenkins/workspace` anyway https://github.com/pytorch/pytorch/blob/main/.github/workflows/_linux-test.yml#L361C19-L361C35

henrylhtsang committed:
Update on "[ci][cutlass backend] Add ci for cutlass backend tests"


redo of https://github.com/pytorch/pytorch/pull/156136

Differential Revision: [D77327309](https://our.internmc.facebook.com/intern/diff/D77327309)

[ghstack-poisoned]

henrylhtsang committed:
Update on "[ci][cutlass backend] Add ci for cutlass backend tests"


redo of https://github.com/pytorch/pytorch/pull/156136

Differential Revision: [D77327309](https://our.internmc.facebook.com/intern/diff/D77327309)

[ghstack-poisoned]

henrylhtsang committed:
Update on "[ci][cutlass backend] Add ci for cutlass backend tests"


redo of https://github.com/pytorch/pytorch/pull/156136

Differential Revision: [D77327309](https://our.internmc.facebook.com/intern/diff/D77327309)

[ghstack-poisoned]

henrylhtsang committed:
Update on "[ci][cutlass backend] Add ci for cutlass backend tests"


redo of https://github.com/pytorch/pytorch/pull/156136

Differential Revision: [D77327309](https://our.internmc.facebook.com/intern/diff/D77327309)

[ghstack-poisoned]

henrylhtsang committed:
Update on "[ci][cutlass backend] Add ci for cutlass backend tests"


redo of https://github.com/pytorch/pytorch/pull/156136

Differential Revision: [D77327309](https://our.internmc.facebook.com/intern/diff/D77327309)

I want to try land the full version first. If the ci is taking too long, we can revert back to only testing for a few names.
```
 -k 'test_max_autotune_cutlass_backend_regular_mm and not test_max_autotune_cutlass_backend_regular_mm_streamk' 
```

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

henrylhtsang committed:
Update on "[ci][cutlass backend] Add ci for cutlass backend tests"


redo of https://github.com/pytorch/pytorch/pull/156136

Differential Revision: [D77327309](https://our.internmc.facebook.com/intern/diff/D77327309)

I want to try land the full version first. If the ci is taking too long, we can revert back to only testing for a few names.
```
 -k 'test_max_autotune_cutlass_backend_regular_mm and not test_max_autotune_cutlass_backend_regular_mm_streamk' 
```

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

henrylhtsang committed:
Update on "[ci][cutlass backend] Add ci for cutlass backend tests"


redo of https://github.com/pytorch/pytorch/pull/156136

Differential Revision: [D77327309](https://our.internmc.facebook.com/intern/diff/D77327309)

I want to try land the full version first. If the ci is taking too long, we can revert back to only testing for a few names.
```
 -k 'test_max_autotune_cutlass_backend_regular_mm and not test_max_autotune_cutlass_backend_regular_mm_streamk' 
```

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

henrylhtsang committed:
Update on "[ci][cutlass backend] Add ci for cutlass backend tests"


redo of https://github.com/pytorch/pytorch/pull/156136

Differential Revision: [D77327309](https://our.internmc.facebook.com/intern/diff/D77327309)

I want to try land the full version first. If the ci is taking too long, we can revert back to only testing for a few names.
```
 -k 'test_max_autotune_cutlass_backend_regular_mm and not test_max_autotune_cutlass_backend_regular_mm_streamk' 
```

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

henrylhtsang committed:
Update on "[ci][cutlass backend] Add ci for cutlass backend tests"


redo of https://github.com/pytorch/pytorch/pull/156136

Differential Revision: [D77327309](https://our.internmc.facebook.com/intern/diff/D77327309)

I want to try land the full version first. If the ci is taking too long, we can revert back to only testing for a few names.
```
 -k 'test_max_autotune_cutlass_backend_regular_mm and not test_max_autotune_cutlass_backend_regular_mm_streamk' 
```

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

huydhn reviewed (approved):
LGTM!

huydhn commented on .github/workflows/h100-cutlass-backend.yml:
```diff
@@ -0,0 +1,55 @@
+      cuda-arch-list: '9.0'
+      test-matrix: |
+        { include: [
+          { config: "h100_cutlass_backend", shard: 1, num_shards: 1, runner: "linux.aws.h100" },
```
In this case, it might be a good idea to set an owner here like https://github.com/pytorch/pytorch/blob/main/.github/workflows/periodic.yml#L94 so that oncall knows who to reach out to when the job fails

huydhn replied:
As you mention about bisecting failures, I think it's a good idea to set an owner like https://github.com/pytorch/pytorch/blob/main/.github/workflows/periodic.yml#L94 so that oncall knows who to reach out to when the job fails

henrylhtsang committed:
Update on "[ci][cutlass backend] Add ci for cutlass backend tests"


redo of https://github.com/pytorch/pytorch/pull/156136

Differential Revision: [D77327309](https://our.internmc.facebook.com/intern/diff/D77327309)

I want to try land the full version first. If the ci is taking too long, we can revert back to only testing for a few names.
```
 -k 'test_max_autotune_cutlass_backend_regular_mm and not test_max_autotune_cutlass_backend_regular_mm_streamk' 
```

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

henrylhtsang committed:
Update on "[ci][cutlass backend] Add ci for cutlass backend tests"


redo of https://github.com/pytorch/pytorch/pull/156136

Differential Revision: [D77327309](https://our.internmc.facebook.com/intern/diff/D77327309)

I want to try land the full version first. If the ci is taking too long, we can revert back to only testing for a few names.
```
 -k 'test_max_autotune_cutlass_backend_regular_mm and not test_max_autotune_cutlass_backend_regular_mm_streamk' 
```

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

henrylhtsang committed:
Update on "[ci][cutlass backend] Add ci for cutlass backend tests"


redo of https://github.com/pytorch/pytorch/pull/156136

Differential Revision: [D77327309](https://our.internmc.facebook.com/intern/diff/D77327309)

I want to try land the full version first. If the ci is taking too long, we can revert back to only testing for a few names.
```
 -k 'test_max_autotune_cutlass_backend_regular_mm and not test_max_autotune_cutlass_backend_regular_mm_streamk' 
```

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

henrylhtsang committed:
Update on "[ci][cutlass backend] Add ci for cutlass backend tests"


redo of https://github.com/pytorch/pytorch/pull/156136

Differential Revision: [D77327309](https://our.internmc.facebook.com/intern/diff/D77327309)

I want to try land the full version first. If the ci is taking too long, we can revert back to only testing for a few names.
```
 -k 'test_max_autotune_cutlass_backend_regular_mm and not test_max_autotune_cutlass_backend_regular_mm_streamk' 
```

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

henrylhtsang commented:
@pytorchbot merge

henrylhtsang commented:
FYI, on cuda 12.8, addmm tests would fail due to accuracy.
